### DISCUSSION TITLE
Pass `req` to all local API operations?

### DISCUSSION THREAD

**Discussion by fabianbru** [payloadcms/payload#2168] [Q&A]
As described in the documentation about the local API for ["Auth Operations"](https://payloadcms.com/docs/local-api/overview#auth-operations) it is possible to pass a `req` express object into the local API call, that will be forwarded to all hooks. The documentation about the CRUD operations is missing this option, but for some of the operations it is also possible to pass in a `req` object.

The "create" operation offers to pass in a `req` object:
```typescript
    await payload.create({
      collection: 'users',
      data: {
        ...
      },
      req: customReq // passing custom req to all hooks is working ✅
    })
```

Unluckily this is only possible for "create", "find", "findVersionByID" or "findById", but not for "delete", "findVersions", "restoreVersion" or "update", e.g.:
```typescript
    await payload.update({
      collection: 'users',
      id: ...
      data: {
        ...
      },
      req: customReq // NOT ALLOWED and non functional ❌
    })
```

Looking at the source, for e.g. ["update"](https://github.com/payloadcms/payload/blob/master/src/collections/operations/local/delete.ts#L11-L19) there is no `req` property like in ["create"](https://github.com/payloadcms/payload/blob/master/src/collections/operations/local/create.ts#L27).

Is this intended, that only read operations are allowed to pass in a `req` object, or is this possible a misalignment that can be fixed to allow all operations to pass a `req` object?

**Comment by DanRibbens** [comment]
I don't think there is a reason for this. If somebody was looking to make a contribution this would make a good PR.
